test/integration_test.py

```diff
@@ -2060,17 +2068,26 @@ def test_response_security_headers(self):
                     else:
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
-                        expected = AzulChaliceApp.security_headers | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
+                        expected = AzulChaliceApp.security_headers() | expected_headers
+                        nonces = set()
+                        for directive in response.headers['Content-Security-Policy'].split(';'):
+                            match = directive_re.fullmatch(directive)
+                            self.assertIsNotNone(match)
+                            name, value = match.groups()
+                            if name in ('script-src', 'style-src'):
+                                nonces.update(nonce_re.findall(value))
```

Comment:
`findall()` is inadequate as it would allow additional strings in between the matches, even strings that would cause the overall value to violate the `source-list` production of the spec.

Since that production is relatively complex, I would be OK with only implementing the non-terminal alternative(s) we actually use, i.e., `nonce-source`, but that can't mean the parser should allow any garbage in between. Rather, the parser should be written as if the RFC read

`source-expression = nonce-source`

instead of 

`source-expression = scheme-source / host-source / keyword-source / nonce-source / hash-source`